idarling/interface/painter.py

```diff
@@ -10,300 +10,148 @@
 
 # You should have received a copy of the GNU General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
-import collections
-import struct
 import sys
 
 import ida_funcs
-import ida_idaapi
 import ida_kernwin
-import ida_nalt
-import ida_registry
-import idautils
+
+from PyQt5.QtCore import (  # noqa: I202
+    QAbstractItemModel,
+    QModelIndex,
+    QObject,
+    Qt,
+)
+from PyQt5.QtGui import QColor
+from PyQt5.QtWidgets import QStyledItemDelegate, QWidget
+import sip
+
+from .widget import StatusWidget
 
 if sys.version_info > (3,):
     long = int
 
 
-class Painter(object):
-    """The painter module is responsible for all the database painting."""
+class Painter(QObject):
+    class ProxyItemDelegate(QStyledItemDelegate):
+        def __init__(self, delegate, parent=None):
+            super(Painter.ProxyItemDelegate, self).__init__(parent)
+            self._delegate = delegate
 
-    DEFCOLOR = 0xffffffff
+        def paint(self, painter, option, index):
+            index = self.parent().model.index(index.row(), index.column())
+            self._delegate.paint(painter, option, index)
 
-    @staticmethod
-    def _get_ida_bg_color():
-        """
-        Return the background color from the disassembly view. It uses the
-        IDA registry to recover the color palettes configured by the user. Then
-        it loads the background color from the currently active palette.
-        """
-        palette = ida_registry.reg_read_binary("Palette")
-        if palette is None:
-            return 0xffffff
-        selected = struct.unpack("<I", palette[8:12])[0]
-        index = 176 + selected * 208
-        return struct.unpack("<I", palette[index : index + 4])[0]  # noqa: E203
+    class ProxyItemModel(QAbstractItemModel):
```

Comment:
Why using QAbstractItemModel rather than QAbstractTableModel ?
Looking at the doc, "When subclassing QAbstractTableModel, you must implement rowCount(), columnCount(), and data(). Default implementations of the index() and parent() functions are provided by QAbstractTableModel.", but for QAbstractItemModel "When subclassing QAbstractItemModel, at the very least you must implement index(), parent(), rowCount(), columnCount(), and data(). These functions are used in all read-only models, and form the basis of editable models."

Reply:
I initially chose it because that's what IDA's UI is subclassing. Using `QAbstractTableModel` doesn't work for me, it crashes on [this assertion](https://github.com/qt/qtbase/blob/5.11/src/widgets/itemviews/qstyleditemdelegate.cpp#L382). Doesn't it for you?